autoreject/autoreject.py

```diff
@@ -1392,3 +1417,68 @@ def plot_epochs(self, epochs, scalings=None, title=''):
             epochs=epochs,
             epoch_colors=epoch_colors, scalings=scalings,
             title='')
+
+    def drop_epochs_with_adjacent_channel_interpolation(
+            self, ch_adjacency, ch_names):
+        """Drop epochs where adjacent channels are marked for interpolation.
+
+        Parameters
+        ----------
+        ch_adjacency : scipy.sparse.csr_matrix, shape (n_channels, n_channels)
+            The adjacency as computed by
+            :func:`mne.channels.find_ch_adjacency`.
+
+        ch_names : list
+            The list of channel names present in adjacency matrix.
+        """
+        bads = np.logical_or(self.labels == 1, self.labels == 2)
+        for idx in range(self.labels.shape[0]):
+            if self.bad_epochs[idx]:  # already bad
+                continue
+            bad_idxs = set(np.where(bads[idx])[0])
+            for bad_idx in bad_idxs:
+                bad_ch = self.ch_names[bad_idx]
+                for bad_idx2 in bad_idxs.difference(set([bad_idx])):
+                    bad_ch2 = self.ch_names[bad_idx2]
+                    if bad_ch in ch_names and bad_ch2 in ch_names:
+                        # adjust to adjacency matrix
+                        bad_idx = ch_names.index(bad_ch)
+                        bad_idx2 = ch_names.index(bad_ch2)
+                        if ch_adjacency[bad_idx, bad_idx2]:
```

Comment:
I feel this should be something like:

"if more than x% of neighboring channels are bad, then drop" ... remember that bad channels still left are actually not that bad because we sort by amplitude before interpolating

Reply:
They're not that bad but they are over the peak-to-peak threshold for that channel...

Reply:
I don't think there is a use-case where you want to keep data where one neighbor is bad but not when two neighbors are bad, it's very uncommon that two neighbors are bad, I think at that point you need to toss the recording.

Reply:
MEG/EEG data is highly correlated. The rank is often much lower than the dimension. You'd be surprised what you can recover by interpolation :) I think the proper approach is to design something based on explained variance

Reply:
you can do outlier detection from the distribution of explained variance for each channel. Again, I think you're missing the point. The goal is to improve the quality of data ... this may come from interpolation, from not dropping a bad epoch (because you drop good channels along with it), or from doing nanmeans, robust regression etc. Unless you quantify somehow it's anyone's guess what an ad hoc method does